File: app/src/main/java/remix/myplayer/misc/tageditor/TagReceiver.java
package remix.myplayer.misc.tageditor;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import remix.myplayer.bean.mp3.Song;
import remix.myplayer.misc.interfaces.OnTagEditListener;

public class TagReceiver extends BroadcastReceiver {
    private final OnTagEditListener mListener;

    public TagReceiver(OnTagEditListener listener){
        mListener = listener;
    }
    @Override
    public void onReceive(Context context, Intent intent) {
        Song newSong = intent.getParcelableExtra("newSong");
        if(mListener != null)
            mListener.onTagEdit(newSong);
    }
}

File: app/src/main/java/remix/myplayer/misc/tageditor/TagReceiver.kt
package remix.myplayer.misc.tageditor

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

import remix.myplayer.bean.mp3.Song
import remix.myplayer.misc.interfaces.OnTagEditListener

class TagReceiver(private val listener: OnTagEditListener?) : BroadcastReceiver() {

  override fun onReceive(context: Context, intent: Intent) {
    val newSong = intent.getParcelableExtra<Song>("newSong")
    listener?.onTagEdit(newSong)
  }

  companion object {
    const val ACTION_EDIT_TAG = "remix.music.ACTION_EDIT_TAG"
  }
}
